iothub/device/src/Message.cs

```diff
@@ -62,20 +62,6 @@ public Message(byte[] byteArray)
             _streamDisposalResponsibility = StreamDisposalResponsibility.Sdk;
         }
 
-        /// <summary>
-        ///
-        /// </summary>
-        /// <param name="payloadCollection"></param>
-        internal Message(PayloadCollection payloadCollection)
-            : this()
-        {
-            ContentEncoding = payloadCollection?.Convention?.PayloadEncoder?.ContentEncoding?.WebName;
-            ContentType = payloadCollection?.Convention?.PayloadSerializer?.ContentType;
```

Comment:
This is where the message properties were getting set before.

Reply:
yeah, the issue with removing the convention from a payload collection context is that the client initialization (where the convention is set) is a completely separate step vs creation of these payload collections.
so what we have now is:
-> create a telemetry message -> no information about convention, so cannot set the ContentEncoding and ContentType.
-> initialize a client instance -> set the convention information
-> call SendTelemetryAsync -> at this point we can set Message.ContentEncoding and Message.ContentType from ClientOptions.PayloadConvention, but that info is still not available in step 1 (cannot be retrieved from a TelemetryMessage object).

This becomes even more problematic with PayloadCollection, since the Add method there needs to convention info while adding a "writable property response", but that info isn't available until (1) a client is initialized, (2) UpdatePropertiesAsync() is called.